### ISSUE TITLE
[$250] Joined thread stuck in LHN with a :red_circle: and it shows up twice in  LHN.

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#47708]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 
**Reproducible in staging?:** Needs Reproduction
**Reproducible in production?:** Needs Reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @AndrewGable 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1724077993452629

## Action Performed:

1. Create a room that userA, userB, and userC are in
2. Post a comment as userA
3. As userB go offline
4. Have userC create a thread on userA's comment, while online
5. Have userB create a thread on userA's comment, while offline
6. Have userB come back online

## Expected Result:
Thread displayed one time without any error 

## Actual Result:
RBR displayed with error and thread displayed twice in LHN
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![Google Chrome 2024-08-19 at 10 32 13](https://github.com/user-attachments/assets/68d0a23c-0948-43e5-9e26-5a0ecdc41e15)
![Google Chrome 2024-08-19 at 10 32 29](https://github.com/user-attachments/assets/a6dfb8bb-5405-40ce-b8c9-62cdb23f7215)

Onyx Data:
```
{
    "reportName": "Hey folks! I've reserved a \"Chef's Table\" style dinner for 6 (can maybe expand to as much as 8) at Brochu's Family Tradition Sunday night at 5:30pm! Early dinner because I'm sure we'll all be knackered from our journeys over. Give a ✋ if you're interested in joining!",
    "isOptimisticReport": true,
    "type": "chat",
    "chatType": "policyRoom",
    "isOwnPolicyExpenseChat": false,
    "isPinned": false,
    "lastActorAccountID": 0,
    "lastMessageTranslationKey": "",
    "lastMessageHtml": "",
    "lastReadTime": "2024-08-15 22:13:10.187",
    "lastVisibleActionCreated": "2024-08-15 22:13:10.187",
    "notificationPreference": "hidden",
    "oldPolicyName": "Expensify Chat",
    "ownerAccountID": 0,
    "parentReportActionID": "1849499882279174199",
    "parentReportID": "4569022691530677",
    "participants": {
        "3064822": {
            "hidden": false,
            "role": "admin"
        },
        "3079111": {
            "hidden": false,
            "role": "member"
        }
    },
    "policyID": "063B20AF6F45E248",
    "reportID": "8214043543876468",
    "stateNum": 0,
    "statusNum": 0,
    "description": "",
    "avatarUrl": "",
    "pendingFields": {
        "createChat": "add"
    },
    "errorFields": {
        "createChat": {
            "1723759990438375": "Auth OpenReport returned an error"
        }
    }
}
```
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ad3492834b7f724d</li>
        <li>Upwork Job ID: 1827073269001925896</li>
        <li>Last Price Increase: 2024-11-29</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by AndrewGable** [comment]
@mallenexpensify - Can you try to reproduce based on those steps?

**Comment by mallenexpensify** [comment]
Was partially able to reproduce, which I think is the core bug. ie. After coming back online as User B, I see RBR in the thread and in LHN (I don't see it shown as twice in LHN though)

I was thinking of adding `External` and asking a C+ to attempt reproduction too, whatcha think?

<img width="835" alt="image" src="https://github.com/user-attachments/assets/000d5161-8d00-487d-b6b8-c34de2ae3e79">

**Comment by AndrewGable** [comment]
Sounds great!

**Comment by mallenexpensify** [comment]
@brunovjk can you plz attempt reproduction and share you're findings?

**Comment by brunovjk** [comment]
> Was partially able to reproduce, which I think is the core bug. ie. After coming back online as User B, I see RBR in the thread and in LHN (I don't see it shown as twice in LHN though)

I got the same result:
![Screenshot 2024-08-24 at 09 09 33](https://github.com/user-attachments/assets/adcc0a0e-e82f-4252-b4b2-5a651a914e09)

Can we specify the expected behavior a bit? Should both messages go to a single thread? In that case, 'message from B' will go to thread from C. I'll investigate further during the week, and I'm also keeping an eye out for proposals. What do you think @mallenexpensify? Thank you :D

**Comment by mallenexpensify** [comment]
@AndrewGable , can you retest to see if you're experiencing both `RBR displayed with error and thread displayed twice in LHN` as the actual behaviour? (since @brunovjk and I weren't able to reproduce 'displayed twice'). 

Then, if needed, we can update the OP for actual and expected.  

Curious how easy/tough this bug will be to fix.

**Comment by ijmalik** [comment]
Hi @mallenexpensify, @AndrewGable, @brunovjk,

I’m encountering the same behavior as @mallenexpensify mentioned [here](https://github.com/Expensify/App/issues/47708#issuecomment-2299983815).

When User-B goes online, they receive the following response. It appears to be an issue on the backend.

**OpenReport:**
```
{
    "code": 666,
    "jsonCode": 666,
    "type": "Expensify\\Libs\\Error\\ExpError",
    "UUID": "5CBABCBB-13F3-4535-9AB2-28C50BDDD7A2",
    "message": "Auth OpenReport returned an error",
    "title": "",
    "data": {
        "onyxData": [
            {
                "onyxMethod": "merge",
                "key": "report_8790068615647771",
                "value": {
                    "errorFields": {
                        "createChat": {
                            "1724824587908180": "Auth OpenReport returned an error"
                        }
                    }
                }
            }
        ]
    },
    "htmlMessage": "",
    "onyxData": [
        {
            "onyxMethod": "merge",
            "key": "report_8790068615647771",
            "value": {
                "errorFields": {
                    "createChat": {
                        "1724824587908180": "Auth OpenReport returned an error"
                    }
                }
            }
        }
    ],
    "requestID": "8ba20ee95962d1e0-MCT"
}
```

**AddComment:**
```
{
    "code": 666,
    "jsonCode": 666,
    "type": "Expensify\\Libs\\Error\\ExpError",
    "UUID": "E490CD6D-46BF-4EE2-8B4F-51672557284A",
    "message": "You don't have access to this chat room.",
    "title": "Room not accessible",
    "data": {
        "onyxData": [
            {
                "onyxMethod": "merge",
                "key": "reportActions_8790068615647771",
                "value": {
                    "3613819486451814722": {
                        "errors": {
                            "1724824588672995": "You don't have access to this chat room."
                        }
                    }
                }
            }
        ]
    },
    "htmlMessage": "",
    "onyxData": [
        {
            "onyxMethod": "merge",
            "key": "reportActions_8790068615647771",
            "value": {
                "3613819486451814722": {
                    "errors": {
                        "1724824588672995": "You don't have access to this chat room."
                    }
                }
            }
        }
    ],
    "requestID": "8ba20eee1bb9de43-MCT"
}
```

**Comment by brunovjk** [comment]
Investing

**Comment by brunovjk** [comment]
Investigating

**Comment by brunovjk** [comment]
I investigated, and it does seem to be a response issue. I’ll trigger an internal to review the backend codes for us. 😊

🎀👀🎀 C+ reviewed

**Comment by brunovjk** [comment]
Sorry for the unnecessary tagging above, @AndrewGable. I wasn’t sure if you could assist with the backend codebase.

**Comment by AndrewGable** [comment]
Yes I can, what am I looking for?

**Comment by brunovjk** [comment]
@AndrewGable take a look at this https://github.com/Expensify/App/issues/47708#issuecomment-2315068847 please :D

**Comment by brunovjk** [comment]
Hey @AndrewGable :D did you make any changes to the backend? I just tested it (v9.0.28-0) and it seems like everything works fine now, I can no longer reproduce the issue:

<details>
<summary>reproduce_47708_web</summary>




https://github.com/user-attachments/assets/40f0ba60-3fbc-472f-9683-3b3cb323fcbd



</details>

What about you @mallenexpensify? Thank you.

**Comment by mallenexpensify** [comment]
Unable to reproduce but only cuz, when I get to 
> Have userB create a thread on userA's comment, while offline

When I click to thread the comment, I just get the loading screen. 
<img width="1002" alt="image" src="https://github.com/user-attachments/assets/8e9cee39-b40b-4292-baab-231da4608f8e">


Throwing retest-weekly on this and bumping to weekly

**Comment by muttmuure** [comment]
Removing from project since it's not a UX reliability issue anymore

**Comment by mallenexpensify** [comment]
Waiting for retest

**Comment by brunovjk** [comment]
Waiting for retest

**Comment by brunovjk** [comment]
Not overdue. Waiting for retest

**Comment by brunovjk** [comment]
I just tested again and by following the same steps I can't reproduce the original issue, but a different one: The first offline message from **User B** never arrives, it is lost and a "Join" button appears, even though user B still remains a member of the room. The tests done after this first failure were successful:

<details>
<summary>Test 1 - Web Chrome</summary>



https://github.com/user-attachments/assets/e7b6ed39-45f2-472f-8f70-9b3ad1cc066d

</details>

<details>
<summary>Test 2 - Web Chrome</summary>



https://github.com/user-attachments/assets/014c73ec-759c-4a20-a25a-eb0ea1cbf755

</details>

We will continue to wait for a `re-test` for a more accurate investigation of the issue.

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.
https://github.com/user-attachments/assets/f704e110-b4f9-4a54-870c-7cc045352509

**Comment by brunovjk** [comment]
From my tests I still conclude that the issue here is in BE.

**Comment by brunovjk** [comment]
From my tests, I still conclude that the issue lies in the backend response when a comment from User B is added to a thread from User A while offline, and User C posts to the same thread while online. This seems to cause a sync conflict.

cc: @AndrewGable @mallenexpensify

**Comment by mallenexpensify** [comment]
Thanks @brunovjk (and @mvtglobally for retesting).  @AndrewGable should we make this internal? Hot Pick? I'm going to add to the #quality project.

**Comment by AndrewGable** [comment]
I personally haven't seen this one pop up again

**Comment by mallenexpensify** [comment]
@AndrewGable the repro steps are ones that I don't think you (or I) would encounter often, if ever... 
Do you think it's worth fixing?  If so, do you think it's a backend fix?

**Comment by brunovjk** [comment]
Since this issue came to my attention, I haven't been able to consistently reproduce it or find a reliable set of steps. However, today I managed to reproduce it with the following steps:

**Steps to Reproduce**:
1. Create a new room.
2. Mention both User B and User C, wait for the "invite" button to appear, click it, and wait for the room to show up in the LHN for Users B and C.
3. Post a comment as User A.
4. Have User B go offline.
5. Have User C create a thread on User A’s comment.
6. Have User B also create a thread on User A’s comment, still while offline.
7. Both User B and User C send a message in User A’s comment thread.
8. Have User B come back online.

**Result in My Test**:
- User C's comment on User A's thread posts without any issues, and the LHN behaves as expected.
- From User A's perspective, everything works fine—no errors or duplicate reports, and only User C's comment arrives. User B’s comment never appears.
- On User B's screen, when they come back online, a "Not Found" page appears. If User B refreshes or navigates back, it looks as if they never posted the comment while offline.

Here is the response from the `AddComment` request after User B comes back online:

```
{
    "code": 666,
    "jsonCode": 666,
    "type": "Expensify\\Libs\\Error\\ExpError",
    "UUID": "E490CD6D-46BF-4EE2-8B4F-51672557284A",
    "message": "You don't have access to this chat room.",
    "title": "Room not accessible",
    "data": {
        "onyxData": [
            {
                "onyxMethod": "merge",
                "key": "reportActions_4167634506996915",
                "value": {
                    "8205893435296449884": {
                        "errors": {
                            "1729771507600209": "You don't have access to this chat room."
                        }
                    }
                }
            }
        ]
    },
    "htmlMessage": "",
    "onyxData": [
        {
            "onyxMethod": "merge",
            "key": "reportActions_4167634506996915",
            "value": {
                "8205893435296449884": {
                    "errors": {
                        "1729771507600209": "You don't have access to this chat room."
                    }
                }
            }
        }
    ],
    "requestID": "8d79d54f68183565-CNF"
}
```
**Video**:

https://github.com/user-attachments/assets/5e74c0b0-d244-4d6f-aef3-cfbcb3790c85

**Comment by brunovjk** [comment]
@AndrewGable and @mallenexpensify can you reproduce it following the steps [above](https://github.com/Expensify/App/issues/47708#issuecomment-2435158066)? Thank you very much.

**Comment by mallenexpensify** [comment]
Thanks @brunovjk , I'm out early today and don't have time to test, I'll get to by EOD Tuesday.

**Comment by brunovjk** [comment]
Excuse me @mallenexpensify, did you get a chance to test again? Thank you :D

**Comment by mallenexpensify** [comment]
I totally forgot, sorry @brunovjk 
Shit, I retested using the steps in the OP and not the one you linked to, sorry. 
I got a totally different actual behaviour too, when I came back online as User B I see
<img width="355" alt="SnagitHelper2024 2024-11-01 17 44 53" src="https://github.com/user-attachments/assets/d500dd80-71d7-41bc-80b4-ddaf98a7019b">

and the thread doesn't show in LHN, only the parent chat
<img width="378" alt="SnagitHelper2024 2024-11-01 17 45 25" src="https://github.com/user-attachments/assets/e28ba37c-4ff4-4c27-963f-63e4817dd3ed">


Only error I see in console
<img width="510" alt="SnagitHelper2024 2024-11-01 17 45 48" src="https://github.com/user-attachments/assets/55f9a825-ce78-43fe-a529-532479b950a1">


I gotta run now, will retest again soon (making daily and making me the owner)

**Comment by brunovjk** [comment]
Great, let me know on how I can assit :D

**Comment by mallenexpensify** [comment]
Ok, got the test done!

Here is what I see for User B
<img width="649" alt="image" src="https://github.com/user-attachments/assets/bac77b5a-7a54-4e42-997b-19aca8f51aeb">


Here's what I see for user A
<img width="813" alt="image" src="https://github.com/user-attachments/assets/39362373-c4d8-48c2-8fb7-8962d35e9bcc">

and for User C 
<img width="755" alt="image" src="https://github.com/user-attachments/assets/2d216898-1736-4a0f-a5c2-ef4696581a45">

Your below breakdown looks correct except, for the final step, when I refresh, I still see errors for user B

https://github.com/user-attachments/assets/a8258d0a-12f3-4f9c-9762-64fe979539bb



([from above](https://github.com/Expensify/App/issues/47708#issuecomment-2435158066))
- User C's comment on User A's thread posts without any issues, and the LHN behaves as expected.
- From User A's perspective, everything works fine—no errors or duplicate reports, and only User C's comment arrives. User B’s comment never appears.
- On User B's screen, when they come back online, a "Not Found" page appears. If User B refreshes or navigates back, it looks as if they never posted the comment while offline.

**Comment by brunovjk** [comment]
Great @mallenexpensify! Hey @AndrewGable, can you please take a look at the BE side regarding the issue [confirmed above](https://github.com/Expensify/App/issues/47708#issuecomment-2455973615)? I didn't find anything in FE that could be causing it. Thank you :D

**Comment by AndrewGable** [comment]
@brunovjk - We are beta testing a new way to request Expensify engineers to look into backend bugs, do you mind creating a new issue using the `Open an internal issue for a backend fix` template and cc me?

**Comment by mallenexpensify** [comment]
@johncschuster I'm off the next week, can you please keep an eye on this issue til I'm back? Thx

**Comment by brunovjk** [comment]
> @brunovjk - We are beta testing a new way to request Expensify engineers to look into backend bugs, do you mind creating a new issue using the `Open an internal issue for a backend fix` template and cc me?

Done: https://github.com/Expensify/App/issues/52179

This is my first issue created in the project :rocket:!!! Let me know if something is not right. Thank you very much.

cc: @AndrewGable @mallenexpensify @johncschuster

**Comment by brunovjk** [comment]
Not overdue, [checking backend](https://github.com/Expensify/App/issues/52179#issuecomment-2468686712).

**Comment by johncschuster** [comment]
Still not overdue. We're checking on BE

**Comment by johncschuster** [comment]
bump

**Comment by johncschuster** [comment]
Bumped the linked issue above

**Comment by mallenexpensify** [comment]
Thanks for keeping 👀 on @johncschuster 
I commented on the other issue
- https://github.com/Expensify/App/issues/52179#issuecomment-2484489683

**Comment by brunovjk** [comment]
_Still not overdue. We're checking on BE_

**Comment by brunovjk** [comment]
Not overdue, same as above :D

**Comment by brunovjk** [comment]
I can't reproduce today (28 Nov):
<details>
<summary>new.expensify.com</summary>



https://github.com/user-attachments/assets/b7ea0fd5-cb25-4d26-8963-6da5cf841e72


</details>

<details>
<summary>dev - v9.0.67-9</summary>



https://github.com/user-attachments/assets/55c7d75a-c23f-414b-853b-587895441c3b


</details>

**Comment by brunovjk** [comment]
@mallenexpensify,  

Given that this issue seems primarily backend-related but impacts the frontend, I suggest closing it for now with the option to reopen if necessary.  

To ensure regressions don't occur, I propose adding a Jest unit test to simulate the following scenario:  

- Users interact with a thread while some are offline, and data syncs upon reconnection.  
The test would validate proper thread display and prevent message duplications or errors.  

Please let me know if you need more details or would prefer a different approach.  

Thanks!

**Comment by mallenexpensify** [comment]
Thanks @brunovjk.  Can you add 🎀 👀 🎀 C+ reviewed to a comment to assign an internal engineer to get 👀 before I close?   Unsure if it's backend-related if we want to close or try to find someone internal to work on.  Thx 

Created a test case GH 
- https://github.com/Expensify/Expensify/issues/449565

<img width="640" alt="image" src="https://github.com/user-attachments/assets/149930ff-75da-439a-b132-1f5ae0f54a6f">

**Comment by brunovjk** [comment]
> Thanks @brunovjk. Can you add 🎀 👀 🎀 C+ reviewed to a comment to assign an internal engineer to get 👀 before I close? Unsure if it's backend-related if we want to close or try to find someone internal to work on. 

Thank you @mallenexpensify 

🎀 👀 🎀

**Comment by brunovjk** [comment]
**Update:** Initially, the issue seemed tied to the `... LHN`, but the symptoms changed over time. Two weeks ago, user C's message disappeared when they came online—no errors, just no trace. This suggests the bug occurs when users interact with a thread while some are offline, and data syncs upon reconnection.

I believe it's a backend issue. We opened an ["internal issue"](https://github.com/Expensify/App/issues/52179), but closed it as it seems fixed now. However, new FE symptoms may reappear, so we propose [creating a test](https://github.com/Expensify/App/issues/47708#issuecomment-2512670316) for this scenario.

#### Scenario Recap:
- Create a room with userA, userB, and userC
- Post a comment as userA
- Have userB go offline
- Have userC create a thread on userA’s comment, while online
- Have userB create a thread on userA’s comment, while offline
- Have userB come back online
- Verify the thread displays only once without any errors

Thoughts, @AndrewGable?

**Comment by AndrewGable** [comment]
Looks good to me! 👍

**Comment by mallenexpensify** [comment]
Thanks @brunovjk , it sounds like....

- We want a test case with the steps above, I created one [here](https://github.com/Expensify/Expensify/issues/451190)
- The bug _should_ be fixed so we can close this
- Compensation might be due to @brunovjk based on time you invested here. If you believe so, follow the below [from the C+ process doc](https://docs.google.com/document/d/1BvohU05MTaHnjOD_vwJv_aqDAirv-ChkyRnKCAvOVyQ/edit?tab=t.0#bookmark=id.25bt9k50o02o)
> Provide brief, yet detailed, reasoning for compensation
State the amount they feel they’re due.   Standard percentages are 25, 50 and 100% of the job price.

**Comment by brunovjk** [comment]
Exactly, I care about testing, so that the issue doesn't occur, and closing this issue. We hadn't discussed compensation, but I don't mind receiving it, since we've been tracking and testing this issue for weeks, but I agree with any decision, what I want most is to close this old issue hehe

**Comment by mallenexpensify** [comment]
@brunovjk how about 25%, $75,  since you spent time on the issue and produced the test case steps (but, we didn't have an accepted solution so I don't think 50% is best (based on our process for payments and percentages).

**Comment by brunovjk** [comment]
Perfect :D

**Comment by mallenexpensify** [comment]
@brunovjk can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~021865204166194804492

**Comment by brunovjk** [comment]
> @brunovjk can you please accept the job and reply here once you have? https://www.upwork.com/jobs/~021865204166194804492

Done xD
